### ISSUE TITLE
Crash in plotSpace:shouldHandlePointingDeviceUpEvent:atPoint when zooming with Trackpad

### ISSUE THREAD

**Issue by gilcel** [core-plot/core-plot#394]
Hi,

I think that there is a bug in plotSpace:shouldHandlePointingDeviceUpEvent:atPoint:
First I thought it was a problem in my app, but testing it in the DropPlot example confirmed the bug.

Try the following in the  "DropPlot Mac" example with Core-Plot (master @ ba05b04) example with XCode 10.0 and macOS 10.13 and a Macbook with Trackpad (it happens also with the mouse but not as frequent as with the Trackpad)

- Open the CSV file with DropPlot.app
- Zoom in (several times) the graph while holding / releasing the Trackpad down
- Release the Trackpad and just move your finger on the Trackpad without clicking: the zoom rectangle is drawed (and will stay in the graph) and the following Assertion failure is displayed in Xcode:

```2018-10-22 10:18:06.655905+0200 DropPlot[2363:25896] *** Assertion failure in -[NSEvent clickCount], /BuildRoot/Library/Caches/com.apple.xbs/Sources/AppKit/AppKit-1561.60.100/AppKit.subproj/NSEvent.m:2118
2018-10-22 10:18:06.656531+0200 DropPlot[2363:25896] [General] Invalid message sent to event "NSEvent: type=ScrollWheel loc=(286.969,244.418) time=4164.3 flags=0 win=0x6040001e3400 winNum=288 ctxt=0x0 deltaX=0.000000 deltaY=0.000000 count:0 phase=Ended momentumPhase=None"
2018-10-22 10:18:06.663260+0200 DropPlot[2363:25896] [General] (
	0   CoreFoundation                      0x00007fff3b36566b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x00007fff625bb942 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff3b36b252 +[NSException raise:format:arguments:] + 98
	3   Foundation                          0x00007fff3d4b0340 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 193
	4   AppKit                              0x00007fff38aa59ed -[NSEvent clickCount] + 191
	5   DropPlot                            0x0000000100005a06 -[CPTPlotDocument plotSpace:shouldHandlePointingDeviceUpEvent:atPoint:] + 214
	6   CorePlot                            0x000000010032bd58 -[CPTPlotSpace pointingDeviceUpEvent:atPoint:] + 152
	7   CorePlot                            0x000000010035fbba -[CPTXYPlotSpace pointingDeviceUpEvent:atPoint:] + 154
	8   CorePlot                            0x0000000100327c93 -[CPTGraph pointingDeviceUpEvent:atPoint:] + 1683
	9   CorePlot                            0x000000010039e46b -[CPTGraphHostingView scrollWheel:] + 1851
	10  AppKit                              0x00007fff391a643d -[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:] + 5040
	11  AppKit                              0x00007fff391a4c70 -[NSWindow(NSEventRouting) sendEvent:] + 497
	12  AppKit                              0x00007fff390060d6 -[NSApplication(NSEvent) sendEvent:] + 2110
	13  AppKit                              0x00007fff388668b5 -[NSApplication run] + 812
	14  AppKit                              0x00007fff38835a72 NSApplicationMain + 804
	15  DropPlot                            0x00000001000065e2 main + 34
	16  libdyld.dylib                       0x00007fff631d2015 start + 1
)

```

Any suggestions to fix this ?

**Comment by eskroch** [comment]
This is a bug in the *DropPlot* example app, not in the Core Plot framework. I fixed it by making the following change in the `-plotSpace:shouldHandlePointingDeviceUpEvent:atPoint:` method.

    if ( event.clickCount == 2 ) {

changed to

    if ( (event.type != NSEventTypeScrollWheel) && (event.clickCount == 2) ) {

Scroll wheel events don't have a `clickCount`.

This fix was made on the `release-2.3` branch.

**Comment by gilcel** [comment]
Ok great! Thanks for the quick fix!
